src/Sparrow/Utils/ZstdStream.cs

```diff
@@ -181,67 +177,70 @@ public override Task WriteAsync(byte[] buffer, int offset, int count, Cancellati
 
         public override async ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken = default)
         {
-            if (_disposeOnce.Disposed)
-                throw new ObjectDisposedException("Object was already disposed!");
-
-            while (buffer.Length > 0)
+            using (_disposerLock.EnsureNotDisposed())
```

Comment:
we need to create EnsureNotDisposedAsync method?
The AsyncReaderWriterLock has a full async support.

Reply:
Done